admin/src/core/services/cognito.service.ts

```diff
@@ -0,0 +1,214 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { Amplify, Auth } from "aws-amplify";
+import { AwsCognitoConfig } from "@api-client";
+import { User } from "@utility/security/user";
+import { ConfigService } from "@utility/services/config.service";
+import { getFakeUser } from "./mock-user";
+import type { CognitoUserSession } from "amazon-cognito-identity-js";
+
+@Injectable()
+export class CognitoService {
+  private config: AwsCognitoConfig = {
+    enabled: false,
+    region: "",
+    userPoolsId: "",
+    userPoolWebClientId: "",
+    mandatorySignIn: true,
+    federationTarget: "",
+    domain: "",
+    scope: [],
+    signUpVerificationMethod: "",
+    frontendRedirectBaseUrl: "",
+  };
+  private cognitoAuthToken: object;
+  private loggedOut: string;
+  private fakeUser: User;
+  public initialized: boolean = false;
+
+  constructor(private configService: ConfigService, private http: HttpClient) {}
+
+  private getParameterByName(name) {
+    const url = window.location.href;
+    name = name.replace(/[\[\]]/g, "\\$&");
+    const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+    const results = regex.exec(url);
+    if (!results) {
+      return null;
+    }
+    if (!results[2]) {
+      return "";
+    }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+  }
+
+  /**
+   * See OIDC Attribute Mapping mapping reference:
+   *      https://github.com/bcgov/nr-forests-access-management/wiki/OIDC-Attribute-Mapping
+   * Note, current user data return for 'userData.username' is matched to "cognito:username" on Cognito.
+   * Which isn't what we really want to display. The display username is "custom:idp_username" from token.
+   */
+  private parseToken(authToken: CognitoUserSession): object {
+    const decodedIdToken = authToken.getIdToken().decodePayload();
+    const decodedAccessToken = authToken.getAccessToken().decodePayload();
+    return {
+      id_token: decodedIdToken,
+      access_token: decodedAccessToken,
+    };
+  }
+
+  public async init(): Promise<any> {
+    /*
+        See Aws-Amplify documenation: 
+        https://docs.amplify.aws/lib/auth/social/q/platform/js/
+        https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
+    */
+
+    let url: string =
+      this.configService.getApiBasePath() + "/api/awsCognitoConfig";
+    let data = await this.http
+      .get(url, { observe: "body", responseType: "json" })
+      .toPromise();
+    this.config = data as AwsCognitoConfig;
+
+    const parsedConfig = {
```

Comment:
Not clear what this is trying to do. Is there a reason we can't use the data from the awsCognitoConfig endpoint directly? I assume there's a reason, I just don't have enough time to study this in depth.

Reply:
E.g. not clear why we store the results of the API call at the class level, but convert the result to something that the Amplify library can use. Why not just store the parsedConfig then?

Reply:
See previous resolved comment and code updated. Thanks Basil.